include/boost/url/impl/url_base.ipp

```diff
@@ -1337,6 +1337,166 @@ end() const noexcept ->
     return iterator(v_, true);
 }
 
+auto
+url_base::
+segments_type::
+erase( url_base::segments_type::iterator pos ) noexcept ->
+    iterator
+{
+    auto last = pos;
+    return erase(pos, ++last);
+}
+
+auto
+url_base::
+segments_type::
+erase( url_base::segments_type::iterator first, url_base::segments_type::iterator last ) noexcept ->
+    iterator
+{
+    BOOST_ASSERT(v_ != nullptr);
+    url_base & v = *v_;
```

Comment:
why `v` instead of `v_`?

Reply:
v_ is really this->v_. Consider:

```
struct X { int y; };

void f();

void use( int );

...

void g( X * x )
{
    use(x->y); // read x->y from memory.
    f();
    use(x->y); // x->y is read from memory a second time because f may have changed it through its own pointer.
}
```

Compare to:

```
void g( X * x )
{
    int y = x->y;
    use(y);
    f();
    use(y); // no need to read x->y again even if it has changed (we know it hasn't).
}
```

The idea is to tell the compiler that `v` will not ever point a different object (because it is local and other code has no access to it and therefore can't change it, whereas `v_` may be changed through `this`).

Reply:
Okay that sounds convincing but please put `&` up against the type:
```
url_base& v = *v_;
```